### PR TITLE
Refactor/mapping to first letter lower

### PR DESCRIPTION
**Pull request by inancgumus**
This is getting hairy so a map makes sense to me.

### CODE REVIEW

**Comment by ka3de** [comment]
I don't particularly see the need for this change, but don't have strong opinion against it. Do you think it makes it more readable with a map? From the performance perspective, the switch implementation is better, but the impact is negligible, especially being in a test, so happy with any version.

**Comment by inancgumus** [comment]
@ka3de Yes, I think it's more readable and manageable than a bunch of if statements. It's a small change and I don't have a strong opinion either, can revert it if you guys don't like it, no worries.

File: browser/mapping_test.go [hunk 1]
```diff
@@ -172,16 +172,15 @@ func toFirstLetterLower(s string) string {
...
-		return "jsonValue"
+	special := map[string]string{
+		"JSON":      "json",
+		"JSONValue": "jsonValue",
+		"URL":       "url",
 	}
+	if v, ok := special[s]; ok {
+		return v
+	}
+	return strings.ToLower(s[:1]) + s[1:]
```
Comment (ankur22) [inline browser/mapping_test.go:183]: Isn't there a chance that `s` will be of length `0`?

**Comment by andrewslotin** [comment]
While the performance-wise maps and `switch` statements are pretty much equal, IIRC a method that contains `switch` only is inlined. Is this the same for map?

**Review by ankur22 — APPROVED** [review]

**Comment by inancgumus** [comment]
Hey, @andrewslotin,

> While the performance-wise maps and `switch` statements are pretty much equal, IIRC a method that contains `switch` only is inlined. Is this the same for map?

Yes, it's the same as long as the function's complexity is low.

**Review by ka3de — APPROVED** [review]
LGTM.

**Pull request merged**